src/prefect/variables.py

```diff
@@ -1,10 +1,98 @@
-from typing import Optional
+from typing import List, Optional
 
-from prefect.client.orchestration import PrefectClient
+from prefect._internal.compatibility.deprecated import deprecated_callable
+from prefect.client.schemas.actions import VariableCreate as VariableRequest
+from prefect.client.schemas.actions import VariableUpdate as VariableUpdateRequest
 from prefect.client.utilities import get_or_create_client
 from prefect.utilities.asyncutils import sync_compatible
 
 
+class Variable(VariableRequest):
+    """
+    Variables are named, mutable string values, much like environment variables. Variables are scoped to a Prefect server instance or a single workspace in Prefect Cloud.
+    https://docs.prefect.io/latest/concepts/variables/
+
+    Arguments:
+        name: A string identifying the variable.
+        value: A string that is the value of the variable.
+        tags: A list of strings to associate with the variable.
+    """
+
+    @classmethod
+    @sync_compatible
+    async def set(
+        cls,
+        name: str,
+        value: str,
+        tags: Optional[List[str]] = None,
+        overwrite: bool = False,
+    ) -> Optional[str]:
+        """
+        Sets a new variable. If one exists with the same name, user must pass `overwrite=True`
+        ```
+            from prefect import variables
+
+            @flow
+            def my_flow():
+                var = variables.Variable.set(name="my_var",value="test_value", tags=["hi", "there"], overwrite=True)
+        ```
+        or
+        ```
+            from prefect.variables import Variable
+
+            @flow
+            async def my_flow():
+                var = await Variable.set(name="my_var",value="test_value", tags=["hi", "there"], overwrite=True)
+        ```
+        """
+        client, _ = get_or_create_client()
+        variable = await client.read_variable_by_name(name)
+        var_dict = {"name": name, "value": value}
+        if tags is not None:
+            var_dict["tags"] = tags
+        if variable:
+            if not overwrite:
+                raise ValueError(
+                    "You are attempting to save a variable with a name that is already in use. If you would like to overwrite the values that are saved, then call .set with `overwrite=True`."
+                )
+            else:
+                var = VariableUpdateRequest(**var_dict)
+                await client.update_variable(variable=var)
+                variable = await client.read_variable_by_name(name)
+                return variable
```

Comment:
I don't think you need `else` on line 58 since you're raising in the other case. also I think we could skip line 61 because you've already read `variable` on line 49 I believe?

Reply:
Pushing a change that removes the else, but I think the variable assignment on 61 is necessary as the patch call doesn't return anything and i need to re-get the variable from the server. Does that make sense or am I missing something.

Reply:
ah no that makes sense, not sure how I misread that earlier